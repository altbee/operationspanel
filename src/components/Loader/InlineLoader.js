import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

const InlineLoader = props => {
  const { visible, className } = props;
  return (
    <span className={!visible ? "d-none" : ""}>
      <FontAwesomeIcon
        icon="circle-notch"
        className={`text-primary ${className}`}
        spin
      />
    </span>
  );
};

InlineLoader.propTypes = {
  visible: PropTypes.bool,
  className: PropTypes.string
};

InlineLoader.defaultProps = {
  visible: false,
  className: ""
};

export default InlineLoader;
