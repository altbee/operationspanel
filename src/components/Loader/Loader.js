import React from "react";
import PropTypes from "prop-types";

const Loader = props => {
  const { visible } = props;
  if (!visible) return null;
  return (
    <div className="loader-overlay">
      <div className="loader" />
    </div>
  );
};

Loader.propTypes = {
  visible: PropTypes.bool
};

Loader.defaultProps = {
  visible: false
};

export default Loader;
