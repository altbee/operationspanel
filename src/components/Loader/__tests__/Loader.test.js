import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Loader from "../Loader";

configure({ adapter: new Adapter() });

describe("<Loader />", () => {
  it("accepts props and renders", () => {
    const rendered = shallow(<Loader visible />);
    expect(rendered.find(".loader-overlay").exists()).toBe(true);
  });

  it("does not show with default props", () => {
    const rendered = shallow(<Loader />);
    expect(rendered.find(".loader-overlay").exists()).toBe(false);
  });

  it("matches the snapshot", () => {
    const tree = shallow(<Loader visible />);
    expect(tree).toMatchSnapshot();
  });
});
