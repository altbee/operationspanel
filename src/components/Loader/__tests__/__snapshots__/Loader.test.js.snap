// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> matches the snapshot 1`] = `
<div
  className="loader-overlay"
>
  <div
    className="loader"
  />
</div>
`;
